tests/test_auth.py

```diff
@@ -75,6 +79,134 @@ def test_invalid_token(self, request_mock, token_mock):
         request_mock.post.return_value = post_json_mock
         self.assertEqual(self.backend.authenticate(request=auth_request), None)
 
+    @override_settings(OIDC_ALLOW_UNSECURED_JWT=True)
+    def test_allowed_unsecured_token(self):
+        """Test get payload data using allowed unsecured tokens."""
+        header = force_bytes(json.dumps({'alg': 'none'}))
+        payload = force_bytes(json.dumps({'foo': 'bar'}))
+        signature = ''
+        token = force_bytes('{}.{}.{}'.format(
+            smart_text(b64encode(header)),
+            smart_text(b64encode(payload)),
+            signature
+        ))
+
+        extracted_payload = self.backend.get_payload_data(token, None)
+        self.assertEqual(payload, extracted_payload)
+
+    @override_settings(OIDC_ALLOW_UNSECURED_JWT=False)
+    def test_disallowed_unsecured_token(self):
+        """Test get payload data using disallowed unsecured tokens."""
+        header = force_bytes(json.dumps({'alg': 'none'}))
+        payload = force_bytes(json.dumps({'foo': 'bar'}))
+        signature = ''
+        token = force_bytes('{}.{}.{}'.format(
+            smart_text(b64encode(header)),
+            smart_text(b64encode(payload)),
+            signature
+        ))
+
+        with self.assertRaises(KeyError):
+            self.backend.get_payload_data(token, None)
+
+    @override_settings(OIDC_ALLOW_UNSECURED_JWT=True)
+    def test_allowed_unsecured_secure_token_secure_verified(self):
+        """Test get secured payload data with disallowed security tokens."""
+        header = force_bytes(json.dumps({'alg': 'HS256', 'typ': 'JWT'}))
+        payload = force_bytes(json.dumps({'foo': 'bar'}))
+
+        # Compute signature
+        key = b'mysupersecuretestkey'
+        h = hmac.HMAC(key, hashes.SHA256(), backend=default_backend())
+        msg = '{}.{}'.format(smart_text(b64encode(header)), smart_text(b64encode(payload)))
+        h.update(force_bytes(msg))
+        signature = b64encode(h.finalize())
+
+        token = '{}.{}.{}'.format(
+            smart_text(b64encode(header)),
+            smart_text(b64encode(payload)),
+            smart_text(signature)
+        )
+        token_bytes = force_bytes(token)
+        key_text = smart_text(key)
+        output = self.backend.get_payload_data(token_bytes, key_text)
+        self.assertEqual(output, payload)
+
+    @override_settings(OIDC_ALLOW_UNSECURED_JWT=False)
+    def test_disallowed_unsecured_secure_token_secure_verified(self):
```

Comment:
same here

Reply:
the same goes for the other tests too. I would suggest to tweak it a bit so someone can understand easily what the test is doing.